packages/docs/celo-codebase/protocol/identity/accounts.md

```diff
@@ -0,0 +1,75 @@
+| Account key            | This is the key with the highest level of permissions, and is thus the most sensitive. It can be used to lock and unlock Celo Gold, and authorize vote, validator, and attestation keys. Note that the account key also has all of the permissions of the other keys. |
+| Validator signer key   | This is the key that has permission to register and manage a Validator or Validator Group, and participate in BFT consensus.                                                                                                                                          |
+| Vote signer key        | This key can be used to vote in Validator elections and on-chain governance.                                                                                                                                                                                          |
+| Attestation signer key | This key is used to sign attestations in Celo's lightweight identity protocol.                                                                                                                                                                                        |
+
+Note that account and signer keys must be unique and may not be reused.
+
+# Key Rotation
+
+Loss of an authorized key no longer has catastrophic implications, as the Account key can just authorize another key in that case. In general, it is good hygiene to rotate your keys, just like you should rotate your passwords.
+
+### Key Rotation for Consensus
+
+Key rotation for Consensus is a bit trickier. Let's assume that a validator is currently elected, actively signing consensus messages and wants to rotate their validator signer key. What they should do is the following:
+
+- Bring up a new validator node and create a new Validator signer key (VSK) on it.
+- Authorize the new Validator signer key with the Account key to overwrite the old Validator signer key.
+
+```bash
+celocli account:authorize --from <ACCOUNT_KEY_ADDRESS> --role validator --signer <NEW_VSK_ADDRESS> --signature <PROOF_OF_NEW_VSK_POSSESSION>
+```
+
+**This should also update your ECDSA public key automatically.**
+
+- Update your BLS key used for consensus. Rotating both the ECDSA and BLS keys is recommended.
+
+```bash
+celocli validator:update-bls-public-key --from <NEW_VSK_ADDRESS> --blsKey <NEW_BLS_KEY> --blsPop <PROOF_OF_NEW_BLS_KEY_POSSESSION>
+```
+
+- Your old node will continue to sign consensus messages for the current epoch, but upon the next epoch will find itself no longer authorized.
+- Your new node will find itself unauthorized, but will be authorized to sign as the next epoch begins.
+
+**Waiting for the epoch to flip before bringing down the retired validator node is essential to zero-downtime rotation when elected.**
```

Comment:
I think the terms "flip" and "zero-downtime" may not be immediately obvious, especially with readers who are not native English speakers, so I would reword this. (And add the warning callout)
```suggestion
{% hint style="warning" %}
The newly authorized keys will only take effect in the next epoch, so the node operating with the old key must remain running until the end of the current epoch to avoid downtime.
{% endhint %}
```